x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -154,7 +154,10 @@ public void authorize(Authentication authentication, String action, TransportReq
                 auditTrail.accessGranted(authentication, action, request, new String[] { SystemUser.ROLE_NAME });
                 return;
             }
-            throw denial(authentication, action, request, new String[] { SystemUser.ROLE_NAME });
+            auditTrail.accessDenied(authentication, action, request, new String[] { SystemUser.ROLE_NAME });
+            ElasticsearchSecurityException e = denialException(authentication, action);
+            assert false : e.getMessage();
```

Comment:
Can you comment this `assert` with something about if this occurs during ES tests, then it probably means that the SystemUser privileges need to be updated.